### PR TITLE
feat: liquidity migration

### LINKED ISSUE #11247: SuperchainERC20: Convert implementation
Once the [Liquidity Migration Specs](https://github.com/ethereum-optimism/specs/pull/294) are merged we will begin with the development of the required modifications.

### LINKED ISSUE #11424: Interop: Update OptimismMintableERC20Factory Implementation
Implement the changes for OptimismMintableERC20Factory based on the work done in [specs](https://github.com/orgs/ethereum-optimism/projects/71/views/1?sliceBy%5Bvalue%5D=skeletor-spaceman&pane=issue&itemId=73211544)

### PR DESCRIPTION
**Pull request by agusduha**
**Description**

The Liquidity Migration Specs will cover the required modifications to allowed converting between legacy and `SuperchainERC20` tokens using the `L2StandardBridge`.

Specs: https://github.com/ethereum-optimism/specs/pull/294

**Tests**

Added `L2StandardBridgeInterop.t.sol`
Updated `OptimismMintableERC20Factory.t.sol`

**Additional context**

TODOs:
- Define the correct predeploy address for `OptimismSuperchainERC20Factory`
- Define a proper naming for the interface `MintableAndBurnable` used in `L2StandardBridgeInterop.sol`

**Metadata**

- Closes https://github.com/ethereum-optimism/optimism/issues/11247
- Closes https://github.com/ethereum-optimism/optimism/issues/11424

### CODE REVIEW

**Comment by tynes** [comment]
@agusduha Looks like ci is failing due to the snapshots being out of date. You should be able to run `just snapshots` to update them

**Comment by agusduha** [comment]
@tynes Snapshots fixed!

File: packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+    /// @notice Mock factory deployment
+    function _mockDeployments(address _factory, address _token, address _deployed) internal {
+        _mockAndExpect(
+            _factory, abi.encodeWithSelector(IOptimismERC20Factory.deployments.selector, _token), abi.encode(_deployed)
+        );
+    }
+
+    /// @notice Assume a valid address for fuzzing
+    function _assumeAddress(address _address) internal view {
```
Comment (tynes) [inline packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol]: There is a `forge-std` helper that does this

Comment (tynes) [reply packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol]: https://github.com/foundry-rs/forge-std/blob/bf6606142994b1e47e2882ce0cd477c020d77623/src/StdCheats.sol#L237

File: packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+        // Arrange
+        _setUpSuperToLegacy(_from, _to);
+
+        // Mock the legacy and superchain factory to return `_remoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _to, _remoteToken);
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _from, _remoteToken);
+
+        // Expect the `Converted` event to be emitted
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
```
Comment (tynes) [inline packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol]: nit: `vm.expectEmit(bool,bool,bool,bool,address)` where all are true is the same as `vm.expectEmit(address)`

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Nice code, very clean

**Pull request merged**